addons/sale/models/payment.py

```diff
@@ -61,33 +61,40 @@ def _set_transaction_pending(self):
         super(PaymentTransaction, self)._set_transaction_pending()
 
         for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state == 'draft')
-            sales_orders.force_quotation_send()
+            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
+            # set quote as sent when transaction is pending
+            # but do not send mail because mail will be send based on configuration in setting.
+            sales_orders.with_context(tracking_disable=True).write({'state': 'sent'})
 
             if record.acquirer_id.provider == 'transfer':
                 for so in record.sale_order_ids:
                     so.reference = record._compute_sale_order_reference(so)
+            # send order confirmation mail
+            sales_orders.send_order_confirmation_mail()
```

Comment:
Why not send the emails and then write the state as 'sent', like the code before?

Reply:
@awa-odoo , same as above